syntax = "proto3";
package zab_client;

import "types.proto";
import "google/protobuf/empty.proto";

service ZabClientService {
  rpc WriteTransaction(WriteTransactionRequest) returns (google.protobuf.Empty) {}

  rpc ReadAccount(ReadAccountRequest) returns (ReadAccountResponse) {}

  rpc DebugReadAccount(ReadAccountRequest) returns (ReadAccountResponse) {}

  rpc DebugHasOutstandingTransactions(google.protobuf.Empty) returns (DebugHasOutstandingTransactionsResponse) {}
}

message WriteTransactionRequest {
  zab.BankTransaction transaction = 1;
}

message ReadAccountRequest {
  uint32 account_id = 1;
}

message ReadAccountResponse {
  int32 balance = 1;
}

message DebugHasOutstandingTransactionsResponse {
  bool has_outstanding_transactions = 1;
}